### PR TITLE
Add detailed evaluation

### PR DESCRIPTION
**Pull request by toddbaert**
This section was mostly done, but I've moved/renamed it and added the detailed evaluation methods, and updating to add `await` keywords for more clarity. 

I've highlighted the new content here: https://github.com/open-feature/docs.openfeature.dev/pull/28#discussion_r929216863

[Direct link](https://deploy-preview-28--lucky-creponne-baf9b3.netlify.app/docs/reference/evaluation-api/#detailed-evaluation) to new content in preview

Closes https://github.com/open-feature/docs.openfeature.dev/issues/26

### CODE REVIEW

File: docs/reference/evaluation-api/using-the-evaluation-api.mdx [hunk 1]
```diff
@@ -118,16 +122,84 @@ MyObject objectValue = client.getObjectValue<MyObject>("objectFlag", MyObjectIns
...
-var boolValue = client.GetBooleanValue("boolFlag", false);
+var boolValue = await client.GetBooleanValue("boolFlag", false);
 
 // get a string value
-var stringValue = client.GetStringValue("stringFlag", "default");
+var stringValue = await client.GetStringValue("stringFlag", "default");
 
 // get an integer value
-var intValue = client.GetIntegerValue("intFlag", 1);
+var intValue = await client.GetIntegerValue("intFlag", 1);
```
Comment (toddbaert) [inline docs/reference/evaluation-api/using-the-evaluation-api.mdx:131]: This was misleading, so I'm awaiting it now to make it clear it's a task/promise.

File: docs/reference/evaluation-api/evaluation-api.mdx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+// get details of int evaluation
+var intDetails = await client.GetNumberDetails("intFlag", 1);
+
+// get details of object evaluation
+var objectDetails = await client.GetObjectDetails("objectFlag", myObjectDefaultInstance);
+```
+
+</TabItem>
+</Tabs>
```
Comment (toddbaert) [inline docs/reference/evaluation-api/evaluation-api.mdx:208]: This section is new.

File: docs/reference/evaluation-api/evaluation-api.mdx [hunk 3]
```diff
@@ -0,0 +1,208 @@
+---
+sidebar_position: 1
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Evaluation API
+
+The Evaluation API is the primary component of OpenFeature that _application authors_ interact with. The Evaluation API allows developers to evaluate feature flags to alter control flow and application characteristics. The evluation API
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply docs/reference/evaluation-api/evaluation-api.mdx:10]: Good catch. Copy paste issue with that last sentence fragment.

I also enabled my local spell checker for `.mdx` files.

File: docs/reference/evaluation-api/evaluation-api.mdx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+#### Evaluation Details Structure Fields
+
+| Field                | Description                                          |
+| -------------------- | ---------------------------------------------------- |
+| flag key             | the flag key                                         |
+| value                | the flag value                                       |
+| error code           | a string denoting the nature of error, if applicable |
+| reason (optional\*)  | a string explaining why the flag value was returned  |
+| variant (optional\*) | the variant assocaited with the return flag value    |
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply docs/reference/evaluation-api/evaluation-api.mdx:152]: Can you give me an example? Especially for the small ones?

[the remark this answers is not in the document]
Comment (beeme1mr) [reply docs/reference/evaluation-api/evaluation-api.mdx:152]: Something like this:
- The unique identifier for a feature flag.
- The value return from the flag evaluation.

[the remark this answers is not in the document]
Comment (toddbaert) [reply docs/reference/evaluation-api/evaluation-api.mdx]: Those are not full sentences.

[the remark this answers is not in the document]
Comment (toddbaert) [reply docs/reference/evaluation-api/evaluation-api.mdx]: Added the same content as above without punctuation, agree it's better even if they are sentence fragmetns.

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**